lib/Echidna/Solidity.hs

```diff
@@ -158,6 +163,17 @@ linkLibraries [] = ""
 linkLibraries ls = "--libraries " ++
   iconcatMap (\i x -> concat [x, ":", show $ addrLibrary + toEnum i, ","]) ls
 
+filterMethods :: MonadThrow m => Filter -> NE.NonEmpty SolSignature -> m (NE.NonEmpty SolSignature)
+filterMethods (Whitelist [])  _ = throwM InvalidMethodFilters 
+filterMethods (Whitelist ic) ms = return $ NE.fromList $ NE.filter (\(m, _) -> T.unpack m `elem` ic) ms
+filterMethods (Blacklist ig) ms = return $ NE.fromList $ NE.filter (\(m, _) -> T.unpack m `notElem` ig) ms
+
+fallback :: SolSignature
+fallback = ("",[])
+
+abiOf :: Text -> SolcContract -> NE.NonEmpty SolSignature
+abiOf pref cc = NE.fromList $ fallback : filter (not . isPrefixOf pref . fst) (elems (cc ^. abiMap) <&> \m -> (m ^. methodName, m ^.. methodInputs . traverse . _2))
```

Comment:
let's use `NE..|` and be total, i know `fallback :` ensures construction but still

Reply:
Fixed.